web/src/left_sidebar_navigation_area.ts

```diff
@@ -172,8 +161,8 @@ export function update_dom_with_unread_counts(
         }
         $header.toggleClass("has-only-muted-unreads", show_muted_count);
         $header.toggleClass(
-            "hide-unread-messages-count",
-            should_mask_header_unread_count(show_muted_count, unmuted_count),
+            "hide_unread_counts",
+            settings_data.should_mask_unread_count(show_muted_count, unmuted_count),
```

Comment:
This line seems like the actual change related to your commit message. I think the rest of this commit is related to making `hide_unread_counts` CSS work with the channels, which is also good but should happen in a different commit with its own commit message.

I'd also love if that other commit could include the work I was doing getting inactive channels set up properly, just having the same styling for `.show-inactive-channels.hide_unread_counts` as `.stream-list-subsection-header.hide_unread_count` (but visible without hover as well)